### PR TITLE
LPS-165064 - Created a New Build From an Existing Template

### PR DESCRIPTION
**Pull request by Elvinho**

### CODE REVIEW

**Comment by Elvinho** [comment]
## :fire: GitRay Report 
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.
 ## Jira Tasks 

 <html><details><summary>More details about: <b>LPS-165064 [FE] Create a New Build From an Existing Template</b>.</summary>
              <br />
              <table>
              <tr>
            <th>Title</th>
            <td><a target="_blank" href="https://issues.liferay.com/browse/LPS-165064">[FE] Create a New Build From an Existing Template</a></td>
        </tr>
              <tr>
            <th>Type</th>
            <td>Technical Task</td>
        </tr>
              <tr>
            <th>Priority</th>
            <td>Minor</td>
        </tr>
              <tr>
            <th>Sub-Tasks</th>
            <td>No Sub-Tasks Opened</td>
        </tr>
              <tr>
            <th>Status</th>
            <td>In Progress</td>
        </tr>
              <tr>
            <th>Assignee</th>
            <td>Elvison Victor</td>
        </tr>
              <tr>
            <th>Reporter</th>
            <td>Elvison Victor</td>
        </tr>
                </table>
          </details><details><summary>More details about: <b>LPS-165061 [FE] Create Search Template in Dropdown</b>.</summary>
              <br />
              <table>
              <tr>
            <th>Title</th>
            <td><a target="_blank" href="https://issues.liferay.com/browse/LPS-165061">[FE] Create Search Template in Dropdown</a></td>
        </tr>
              <tr>
            <th>Type</th>
            <td>Technical Task</td>
        </tr>
              <tr>
            <th>Priority</th>
            <td>Minor</td>
        </tr>
              <tr>
            <th>Sub-Tasks</th>
            <td>No Sub-Tasks Opened</td>
        </tr>
              <tr>
            <th>Status</th>
            <td>In Progress</td>
        </tr>
              <tr>
            <th>Assignee</th>
            <td>Elvison Victor</td>
        </tr>
              <tr>
            <th>Reporter</th>
            <td>Elvison Victor</td>
        </tr>
                </table>
          </details></html> 

---

Hey, don't forget to update all related tickets with correct status, such as **In Review** :wink: 
### Git Stats 

8 files changed, 191 insertions(+), 55 deletions(-)

**Comment by Elvinho** [comment]
ci:test:sf

**Comment by rafaelfs98** [comment]
ci:test:sf

**Review by kevenleone — CHANGES_REQUESTED** [review]
x

**Comment by Elvinho** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	);
+
+	const templatesCount = buildResponse?.totalCount || 0;
+	const buildTemplates = buildResponseWithSearch?.items || [];
+
+	const items: Dropdown = [
+		{
+			items: [
+				{
+					label: 'New Build',
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx:75]:
```suggestion
					label: i18n.translate('new-build'),
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+					path: './create',
+				},
+				{
+					label: 'New Template',
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx:79]:
```suggestion
					label: i18n.translate('new-template'),
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+	);
+
+	const {data: buildResponse} = useFetch<APIResponse<TestrayBuild>>(
+		`${testrayBuildImpl.resource}&filter=${totalFilter}&fields=id`
+	);
+
+	const templatesCount = buildResponse?.totalCount || 0;
+	const buildTemplates = buildResponseWithSearch?.items || [];
+
+	const items: Dropdown = [
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx:71]:
```suggestion
	const dropdownItems: Dropdown = [
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+	const {data: buildResponseWithSearch} = useFetch<APIResponse<TestrayBuild>>(
+		`${testrayBuildImpl.resource}&filter=${searchFilter}`
+	);
+
+	const {data: buildResponse} = useFetch<APIResponse<TestrayBuild>>(
+		`${testrayBuildImpl.resource}&filter=${totalFilter}&fields=id`
+	);
+
+	const templatesCount = buildResponse?.totalCount || 0;
+	const buildTemplates = buildResponseWithSearch?.items || [];
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx:69]:
```suggestion
        const buildTemplates = buildResponseWithSearch?.items || [];
	const templatesCount = buildResponse?.totalCount || 0;
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+import {SearchBuilder} from '../../../../util/search';
+
+type BuildAddButtonProps = {
+	routineId: string;
+};
+
+const BuildAddButton: React.FC<BuildAddButtonProps> = ({routineId}) => {
+	const navigate = useNavigate();
+	const [searchTemplate, setSearchTemplate] = useState('');
+	const [active, setActive] = useState(false);
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx:37]: Order and Group

```suggestion
	const navigate = useNavigate();
	
	const [active, setActive] = useState(false);
	const [searchTemplate, setSearchTemplate] = useState('');
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+
+	const searchFilter = baseFilter
+		.and()
+		.contains('name', debouncedValue)
+		.build();
+
+	const totalFilter = baseFilter
+		.and()
+		.contains('name', debouncedValue)
+		.build();
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx:58]:
```suggestion
	const totalFilter = baseFilter.build();
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+										<div className="align-items-center d-flex testray-sidebar-item text-dark">
+											<span className="ml-1 testray-sidebar-text">
+												{label}
+											</span>
+										</div>
+									</ClayDropDown.Item>
+								</React.Fragment>
+							))}
+
+							{!!buildTemplates && (
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx:128]:
```suggestion
							{!!buildTemplates.length && (
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+								symbol="plus"
+							/>
+						</div>
+					</Tooltip>
+				</div>
+			}
+		>
+			<ClayDropDown.ItemList>
+				{items.map((section, index) => (
+					<div key={index}>
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Project/Routines/Builds/BuildAddButton.tsx:108]: try to remove this div

**Review by kevenleone — CHANGES_REQUESTED** [review]

**Comment by Elvinho** [comment]
ci:test:sf

**Comment by kevenleone** [comment]
ci:forward

**Pull request closed without merging**